### ISSUE TITLE
Database setup boilerplate

### ISSUE THREAD

**Issue by oliverjam** [foundersandcoders/master-reference#989]
I've just gone through all the Week 6 database setup stuff (wrote my first SQL in 2 years 😍!) and it feels like we could clean it up a fair bit.

There's a lot of magic boilerplate that it looks like teams copy from project to project. Every project ends up with a `db_connection.js` that does a load of string parsing of their connection URL.

```js
const { Pool } = require('pg');
const url = require('url');

const params = url.parse(process.env.DB_URL);
const [username, password] = params.auth.split(':');

const options = {
  host: params.hostname,
  port: params.port,
  database: params.pathname.split('/')[1],
  max: process.env.DB_MAX_CONNECTIONS || 2,
  user: username,
  password,
  ssl: params.hostname !== 'localhost'
};

module.exports = new Pool(options);
```

Most of this isn't actually necessary: `pg` [supports standard Postgres environment variables](https://node-postgres.com/features/connecting), so if you set them with the right names `Pool` doesn't need any options:

```sh
PGUSER=dbuser
PGHOST=database.server.com
PGPASSWORD=secretpassword
PGDATABASE=mydb
PGPORT=3211
```

```js
const { Pool } = require('pg');

module.exports = new Pool();
```

_Even simpler_, `pg` supports passing the connection URL directly, so we could keep the same env setup and just do:

```js
const { Pool } = require('pg');

const connectionString = process.env.DB_URL;

module.exports = new Pool({ connectionString });
```

I'm not sure who's going to be mentoring the next Database Week, but it might be a good idea to simplify the initial codealong?

**Comment by Oliversw** [comment]
This definitely seems super useful! We just copied the connection file every single week because we didn't think anything could really be changed. Being a bit more concise and deliberate about it would be lovely

**Comment by ali-7** [comment]
yes, also ` Legacy URL API is Deprecated`  >>> `url.parse`

**Comment by oliverjam** [comment]
@ali-7 ah I didn't realise node had implemented the `URL` constructor, cool

**Comment by ali-7** [comment]
@oliverjam yes, it was a new feature since 8v, I think, then they make it permanent

**Comment by mattlub** [comment]
agree, looks good. I guess we lose those `ssl` and `max` options but I'm not even sure if they're valid options (https://www.postgresql.org/docs/9.1/libpq-envars.html)?

**Comment by oliverjam** [comment]
I think they are both valid, no reason not to keep them: https://node-postgres.com/api/client

We can do something like:
```js
const { Pool } = require("pg");

const connectionString = process.env.DB_URL;

module.exports = new Pool({
  connectionString,
  max: 2,
  ssl: !connectionString.includes("localhost")
});
```

**Comment by mattlub** [comment]
true. Do we know why we set `max` to 2? the default is `10`

https://node-postgres.com/api/pool#new-pool-config-object-

**Comment by oliverjam** [comment]
I honestly don't know where half this stuff comes from. One person makes it up in an example and suddenly we've got 7 workshops and 5 cohorts worth of projects copying it 😆 

If it has a sensible default we should leave it out. The less config the better when you're still learning

**Comment by arrested-developer** [comment]
I was just scanning over #991 and I wonder if the convention of using a database URL has come about because this is the way Heroku sets up the Postgres addon?

When we want students to deploy their app via Heroku, they will need to understand what the `DATABASE_URL` environment variable is and how to parse it, whereas if they're already used to setting up their database this way they can relatively painlessly drop the existing app into Heroku. Perhaps more importantly, when they get to the point that they want to develop an app locally, and test on Travis and run on Heroku, it seems to make sense to use the `process.env.DATABASE_URL` across all environments.

Not sure if that's reason enough on its own to _start_ learning about database connections by parsing a URL but it might explain why there are so many examples of parsing a database URL from the environment variables.

**Comment by oliverjam** [comment]
@arrested-developer sorry if my original comment was confusing, but I'm suggesting we keep using the database URL (like the one Heroku gives you). The difference is we don't need to do all the parsing. The `pg` module can do that for you if you just pass the URL in as the `connectionString` option.
